lwgenerate/aws_controltower/aws_controltower.go

```diff
@@ -0,0 +1,455 @@
+	// Validate inputs
+	if err := args.validate(); err != nil {
+		return "", errors.Wrap(err, "invalid inputs")
+	}
+
+	// Create blocks
+	requiredProviders, err := createRequiredProviders()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate required providers")
+	}
+
+	awsProvider, err := createAwsProvider(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate aws provider")
+	}
+
+	laceworkProvider, err := createLaceworkProvider(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate lacework provider")
+	}
+
+	// controlTower
+	controlTowerModule, err := createCloudTrailControlTower(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate aws controlTower module")
+	}
+
+	// Render
+	hclBlocks := lwgenerate.CreateHclStringOutput(
+		lwgenerate.CombineHclBlocks(
+			requiredProviders,
+			awsProvider,
+			laceworkProvider,
+			controlTowerModule,
+		),
+	)
+	return hclBlocks, nil
+}
+
+type AwsSubAccount struct {
+	// The name of the AwsProfile to use (in AWS configuration)
+	AwsProfile string
+
+	// The AwsRegion this profile should use if any resources are created
+	AwsRegion string
+
+	// The Alias of the provider block
+	Alias string
+}
+
+func NewAwsSubAccount(profile string, region string, alias ...string) AwsSubAccount {
+	subaccount := AwsSubAccount{AwsProfile: profile, AwsRegion: region}
+	if len(alias) > 0 {
+		subaccount.Alias = alias[0]
+	}
+	return subaccount
+}
+
+func (args GenerateAwsControlTowerTfConfigurationArgs) validate() error {
+	return nil
+}
+
+type AwsControlTowerTerraformModifier func(c *GenerateAwsControlTowerTfConfigurationArgs)
+
+// NewTerraform returns an instance of the GenerateAwsControlTowerTfConfigurationArgs struct.
+//
+// Note: Additional configuration details may be set using modifiers of the AwsControlTowerTerraformModifier type
+//
+// Basic usage: Initialize a new AwsControlTowerTerraformModifier struct, with a non-default AWS profile set. Then
+// use generate to create a string output of the required HCL.
+//
+//	hcl, err := aws_controltower.NewTerraform("us-east-1")
+//	  .WithAwsProfile("mycorp-profile")).Generate()
+func NewTerraform(s3BucketArn string, snsTopicArn string,
+	mods ...AwsControlTowerTerraformModifier) *GenerateAwsControlTowerTfConfigurationArgs {
+	config := &GenerateAwsControlTowerTfConfigurationArgs{
+		S3BucketArn: s3BucketArn,
+		SNSTopicArn: snsTopicArn,
+	}
+	for _, m := range mods {
+		m(config)
+	}
+	return config
+}
+
+func WithCrossAccountPolicyName(name string) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.CrossAccountPolicyName = name
+	}
+}
+
+func WithLaceworkAccountID(account string) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.LaceworkAccountID = account
+	}
+}
+
+func WithLaceworkProfile(profile string) AwsControlTowerTerraformModifier {
```

Comment:
The functions from line 213->243 are unused.  The variables don't look to be set anywhere.  Why are these options not used?